assets/src/components/vehiclePropertiesCard.tsx

```diff
@@ -0,0 +1,191 @@
+}: {
+  timestamp: number
+}): React.ReactElement => {
+  const epochNowInSeconds = useCurrentTimeSeconds()
+  return (
+    <output
+      aria-label="Time Since Last Update Received"
+      className="data-stale-time label font-xs-reg"
+    >
+      Updated {epochNowInSeconds - timestamp} sec ago
+    </output>
+  )
+}
+
+const VehicleDataStaleTime = ({ vehicle }: VehicleProp): React.ReactElement => (
+  <DataStaleTime timestamp={vehicle.timestamp} />
+)
+// #endregion
+
+// #region Vehicle Work Info
+enum HideSensitiveInfo {
+  All,
+  Value,
+  None,
+}
+
+// Table Row Name Value Pair Data
+interface TrNameValueProps {
+  name: string
+  children: ReactNode
+  idPrefix?: string
+  sensitivity?: HideSensitiveInfo
+}
+
+const maskClass = "fs-mask"
+const shouldMaskInfo = (value: boolean) => value && maskClass
+
+// Table Row Name Value Pair
+const TrNameValue = ({
+  name,
+  children: value,
+  idPrefix,
+  sensitivity: sensitive = HideSensitiveInfo.None,
+}: TrNameValueProps): React.ReactElement => {
+  const id = (idPrefix ?? name) + useId()
+  return (
+    <tr className={shouldMaskInfo(sensitive === HideSensitiveInfo.All) || ""}>
+      <th className="kv-key font-s-semi" scope="row" id={id}>
+        {name}
+      </th>
+      <td
+        className={classNames([
+          "kv-value font-s-reg",
+          shouldMaskInfo(sensitive !== HideSensitiveInfo.None),
+        ])}
+        aria-labelledby={id}
+      >
+        {value}
+      </td>
+    </tr>
+  )
+}
+
+const VehicleWorkInfo = ({ vehicle }: VehicleProp): React.ReactElement => (
+  <>
+    <table className="m-vehicle-work-info">
+      <tbody className="m-vehicle-work-info__items">
+        <TrNameValue name="run">{vehicle.runId ?? "N/A"}</TrNameValue>
+        <TrNameValue name="vehicle">{vehicle.label ?? "N/A"}</TrNameValue>
+        <TrNameValue name="operator" sensitivity={HideSensitiveInfo.All}>
+          {classNames([
+            vehicle.operatorFirstName,
+            vehicle.operatorLastName,
+            vehicle.operatorId ? `#${vehicle.operatorId}` : null,
+          ]) || "N/A"}
+        </TrNameValue>
+      </tbody>
+    </table>
+  </>
+)
+// #endregion
+
+// #region Vehicle Location
+const CurrentLocation = ({ vehicle }: VehicleProp): React.ReactElement => {
+  const intersection = useNearestIntersection(
+    vehicle.latitude,
+    vehicle.longitude
+  )
+  return <>{intersection ?? "Exact location cannot be determined"}</>
+}
+
+const VehicleNearestIntersection = ({
+  vehicle,
+}: VehicleProp): React.ReactElement => {
+  const id = `current-location-${useId()}`
+  return (
+    <div className="m-current-location">
+      <label
+        className="m-current-location__label label font-xs-reg title-case"
+        htmlFor={id}
+      >
+        Current Location
+      </label>
+      <output className="m-current-location__value label font-s-semi" id={id}>
+        <CurrentLocation vehicle={vehicle} />
+      </output>
+    </div>
+  )
+}
+
+// #endregion
```

Comment:
What I did find from my work on implementing loading state was that the component with the `key` value will be discarded and rerendered, which means for react testing library, you need to do `screen.getByRole` again to get the updated value. 

But this actually is a test working correctly, which is that the DOM element should not be discarded, and instead the `key` value should go on the `<CurrentLocation>` element and not it's parent `<output>` element which is the target of `screen.getByRole("status", { name: "Current Location"})`.